src/crewai/crew.py

```diff
@@ -278,14 +283,16 @@ def kickoff(
             raise NotImplementedError(
                 f"The process '{self.process}' is not implemented yet."
             )
-
         metrics = metrics + [
-            agent._token_process.get_summary() for agent in self.agents
+            agent.token_process.get_summary()
+            for agent in self.agents
+            if hasattr(agent, "token_process")
```

Comment:
Same here, but with another protocol name, like `AgentWithTokenization` or something like that.

Reply:
actually just removed all of the hasattr checks for token_process. it makes sense for all agents to require this - so required with BaseAgent to provide that. And uses the TokenProcess class for it.